src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php

```diff
@@ -46,6 +46,13 @@ protected function createController($controller)
             $resolvedController[0] = $this->configureController($resolvedController[0]);
         }
 
+        if (\is_object($resolvedController) && method_exists($resolvedController, '__invoke')) {
+            $resolvedController = [
+                $this->configureController($resolvedController),
+                '__invoke',
+            ];
```

Comment:
@rjwebdev You don't need to make it an array since `$resolvedController` is already a callable, passing it through `configureController` should be enough:
```php
        if (\is_object($resolvedController) && method_exists($resolvedController, '__invoke')) {
            $this->configureController($resolvedController);
        }
```

Reply:
As far as I know, I should make an array out of it because both the class name and the method are expected in the resolver.

Reply:
Actually, the controller resolver needs to return any kind of callable, doesn't have to be an array. This is the reason the `ControllerResolverTest::testGetControllerInvokableService` is failing.

Also, this change doesn't seem to be necessary on the master branch as this case already works there.

Reply:
I agree on that one but the problem is (from where I started this PR) that no container was injected in the controller when the service id belonging to the controller was used in the routing configuration without defining a method. According to the sf docs, when autowiring is used, the __invoke method should be called: https://symfony.com/doc/master/controller/service.html#invokable-controllers

Reply:
Yes, that's why you need to call `configureController`, but that has nothing to do with making it into an array. I just tried it and it works fine with the example code from earlier.